pystac/stac_object.py

```diff
@@ -419,9 +419,12 @@ def from_file(cls, href):
 
         # If this is a root catalog, set the root to the catalog instance.
         root_link = o.get_root_link()
-        if not root_link.is_resolved():
-            if root_link.get_absolute_href() == href:
-                o.set_root(o, link_type=root_link.link_type)
+
+        if root_link is not None:
```

Comment:
Is this a fix to something you hit on? Wondering if we should make this fix independently as I think we'll release a 0.8 bugfix version before the 0.9 upgrade.

Reply:
Ok, I remembered what the issue was here. This was erroring out the unit tests [here](https://github.com/azavea/pystac/blob/b318a24b89e8854f7f80a4e0d71217acad2b5795/tests/test_item.py#L61) when it  tried to create an item from [this file](https://github.com/azavea/pystac/blob/sjk/interactive-rebase-0.9/tests/data-files/examples/0.9.0/item-spec/examples/datetimerange.json). It doesn't have a `root_link`.

You're right, this seems like a general bug fix so I can put it in the bug fix PR

Reply:
on second thought, reverting this change will cause the CI to fail so maybe it makes sense to keep it in. I guess it depends on the order that the branches get merged. I will hold off on changing this for now. Let me know what course of action you think would be best.